compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -66,6 +66,9 @@ fn has_significant_drop_raw<'tcx>(
 struct NeedsDropTypes<'tcx, F> {
     tcx: TyCtxt<'tcx>,
     param_env: ty::ParamEnv<'tcx>,
+    // Whether to reveal coroutine witnesses, this is set
+    // to `false` unless we compute `needs_drop` for a generator witness.
```

Comment:
Nit: there are a couple remaining mentions of "generators"

```suggestion
    // to `false` unless we compute `needs_drop` for a coroutine witness.
```